scanapp/tasks.py

```diff
@@ -57,6 +63,44 @@ def scan_code_async(url, scan_id, path, file_name):
 
 
 @app.task
+def scan_code_async_final(url, scan_id, path):
+    """
+    Create and initialise the git repository at a certain path and clone the git repo using the 'url'
+    and then get the scan done asynchronously.
+    """
+    logger = logging.getLogger(__name__)
+    logger.info('git repo detected')
+
+    home_path = expanduser("~")
+    os.chdir(home_path)
+
+    os.chdir(path)
+
+    print os.curdir
+
+    clean_url = ''.join(e for e in url if e.isalnum())
+    dir_name = clean_url
+
+    if os.path.isdir(dir_name):
+        shutil.rmtree(dir_name)
+
+    os.mkdir(dir_name)
```

Comment:
The whole way you handle directories is broken.
You are doing a checkout. You want to get a temp directory and nothing else. Use the scancode.commoncode.fileutils code at https://github.com/nexB/scancode-toolkit/blob/22bf88df19768dd207973ed2db91249822a0c446/src/commoncode/fileutils.py#L120 and use a well defined base dir. Each new clone should go in a new temp directory. Never deleted